plot-lib/plot/private/plot2d/plot-area.rkt

```diff
@@ -355,7 +355,7 @@
 
     (: get-x-tick-label-params (-> (Listof Label-Params)))
     (define (get-x-tick-label-params)
-      (if (plot-x-axis?)
+      (if (and (plot-x-axis?) (plot-x-tick-labels?))
```

Comment:
We might need a new variable, `draw-x-tick-labels?`, similar to `draw-x-far-tick-labels?` as the code seems to assume that they are always drawn (see `max-x-tick-label-height`, `get-x-tick-label-params`),  this line should be

```racket
(if (and (plot-x-axis?) draw-x-tick-labels?) ...)
```

... and the other places updated too.